src/components/scenes/FioPendingRequestDetailsScene.js

```diff
@@ -0,0 +1,316 @@
+    // some complex logic because 'toCurrencyCode/fromCurrencyCode'
+    // can be denomination (needs to change to actual currencyCode)
+    if (toWallet) {
+      if (toWallet.enabledTokens.length > 1) {
+        // could be token
+        excludedCurrencyCode = toCurrencyCode
+      } else {
+        excludedCurrencyCode = toWallet.currencyCode
+      }
+    }
+    const walletCurrencyCodes = []
+    const allowedWallets = []
+    for (const id in wallets) {
+      const wallet = wallets[id]
+      if (wallet.currencyCode === 'ETH' && wallet.enabledTokens.length > 0) {
+        walletCurrencyCodes.push(wallet.currencyCode)
+        if (wallet.receiveAddress && wallet.receiveAddress.publicAddress) {
+          if (wallet.currencyCode === this.props.selectedFioPendingRequest.content.chain_code) {
+            allowedWallets.push(wallets[id])
+          }
+        }
+      }
+      if (excludedCurrencyCode !== wallet.currencyCode) {
+        walletCurrencyCodes.push(wallet.currencyCode)
+        if (wallet.receiveAddress && wallet.receiveAddress.publicAddress) {
+          if (wallet.currencyCode === this.props.selectedFioPendingRequest.content.chain_code) {
+            allowedWallets.push(wallets[id])
+          }
+        }
+      }
+    }
+    const supportedWalletTypes = []
+    for (let i = 0; i < this.props.supportedWalletTypes.length; i++) {
+      const swt = this.props.supportedWalletTypes[i]
+      if (!walletCurrencyCodes.includes(swt.currencyCode) && swt.currencyCode !== 'EOS' && excludedCurrencyCode !== swt.currencyCode) {
+        supportedWalletTypes.push(swt)
+      }
+    }
+
+    Airship.show(bridge => (
+      <WalletListModal
+        bridge={bridge}
+        wallets={allowedWallets}
+        type={Constants.FROM}
+        existingWalletToFilterId={toWallet.id}
+        existingWalletToFilterCurrencyCode={toCurrencyCode}
+        supportedWalletTypes={supportedWalletTypes}
+        excludedCurrencyCode={[]}
+        showWalletCreators={false}
+        state={this.props.state}
+        headerTitle={s.strings.fio_src_wallet}
+        excludedTokens={[]}
+        noWalletCodes={[]}
+        disableZeroBalance={false}
+      />
+    )).then((response: any) => {
+      if (response) {
+        if (response.id) {
+          onSelectWallet(response.id, response.currencyCode)
+          return
+        }
+        this.props.createCurrencyWallet(response.value, response.currencyCode)
+      }
+    })
+    return null
+  }
+
+  render () {
+    const materialStyle = MaterialInput
+    materialStyle.tintColor = 'white'
+    materialStyle.baseColor = 'white'
+
+    return (
+      <SceneWrapper>
+        <SafeAreaView>
+          <View>
+            {this.amountField(
+              this.props.selectedFioPendingRequest.content.amount,
+              this.props.selectedFioPendingRequest.content.token_code,
+              this.props.selectedFioPendingRequest.content.token_code,
+              '$'
```

Comment:
`$`Should not be hardcoded. Please use the Fiat symbol from the account.

Reply:
yes, fixed